File: Q003.py
# 2021-03-10
a, b, c = map(int, input().split())

alist = []
blist = []
for i in range(1, a+1):
    if a%i == 0:
        alist.append(i)

for i in range(1, b+1):
    if b%i == 0:
        blist.append(i)

ans = 0
for i in range(len(alist)):
    for j in range(len(blist)):
        if alist[i] + blist[j] <= c:
            ans += 1

print(ans)

File: Q003.cpp
// 2020-07-16

#include <bits/stdc++.h>
#define pb push_back
#define all(x) x.begin(), x.end()
using namespace std;
int main(){
ios::sync_with_stdio(0);
cin.tie(0);
    int a, b, x;
    cin >> a >> b >> x;
    vector<int> alist, blist;
    for(int i=1;i<=a;i++) {
        if (a%i == 0) alist.pb(i);
    }
    for(int i=1;i<=b;i++) {
        if (b%i == 0) blist.pb(i);
    }
    int ans=0;
    for(int i=0;i<alist.size();i++){
        for(int j=0;j<blist.size();j++){
            if (alist[i] + blist[j] <= x) ans++;
        }
    }
    cout << ans;

}
